### PR TITLE
Non numerical inputs

### PR DESCRIPTION
**Pull request by gksander**
This PR is kind of fat, but the goal is pretty focused: support non-numerical inputs for `xKey`. Actually only supporting `string | number` now, because those are the values that Reanimated will serialize appropriately – but the addition of `string` is very useful, especially for e.g. bar charts or dates.

## API changes

The core API change is the signature of `useChartPressState`. Instead of `useChartPressState(["highTmp"])` you now pass an initialization object like:

```ts
useChartPressState({ x: 0, y: { highTmp: 0 } });
```

which sets initial values for state variables. This is nice because you can explicitly set initial values for the state variables, but it was also needed for type-inference stuff (primarily: knowing what type `x` will be, either `number` or `string`).

## Code changes herein

The code changes then consist of:

- Updating `useChartPressState` as mentioned above.
- Updating a looooot of TS generics to support things other than numbers, but the end-result is nicer IMO.
- Small changes to `transformInputData.ts` and `CartesianChart.tsx` to take into account string input.

### CODE REVIEW

File: example/app/scatter.tsx [hunk 1]
```diff
@@ -71,6 +79,14 @@ export default function ScatterPage(props: { segment: string }) {
...
                     color="black"
                   />
+                  {isActive && (
+                    <Text
+                      x={state.x.position}
+                      y={state.y.listenCount.position}
+                      text={state.x.value}
+                      font={font}
+                    />
+                  )}
```
Comment (gksander) [inline example/app/scatter.tsx:89]: This was initially for testing out the `SharedValue<string>` for the update. I forgot to remove this, but maybe we can leave it in? idk, it's not prettied up or anything.

File: example/app/ordinal-data.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ const colors = [appColors.tint, "#818cf8"];
 export default function OrdinalDataScreen(props: { segment: string }) {
   const description = descriptionForRoute(props.segment);
   const font = useFont(inter, 12);
-  const { state: activePress } = useChartPressState(["high"]);
+  const { state: activePress } = useChartPressState({ x: 0, y: { high: 0 } });
```
[the remark this answers is not in the document]
Comment (gksander) [reply example/app/ordinal-data.tsx:30]: Those are "initial values" for those fields, the values that'll sit in the `SharedValue` before the first press occurs.

[the remark this answers is not in the document]
Comment (carloskelly13) [reply example/app/ordinal-data.tsx:30]: I see, you need to know the shape by what's passed in, to correctly infer the types (numeric vs string).

[the remark this answers is not in the document]
Comment (carloskelly13) [reply example/app/ordinal-data.tsx:30]: backing out, we're passing the return value of a hook into our component. are we doing too much? should we be returning the active press state from `CartesianChart` using an `onChartPress` prop or imperative handle? Then we could likely know the shape/types that way.

[the remark this answers is not in the document]
Comment (gksander) [reply example/app/ordinal-data.tsx:30]: > are we doing too much?

_Maybe_. But somehow we need to generate a bunch of `SharedValue`s for the user to use wherever they want, but let our chart have access to them. We could maybe create those in the `CartesianChart`, but I'm not convinced the imperative handle approach is going to be any less ugly than this from the end-user's perspective. Any ideas what that might look like?

As for `onChartPress`, if you want to stay on UI thread and use `SharedValue`s then the `onCharPress` would have to pass back effectively the same thing the `useChartPressState` is creating, but now the user has to create a variable and intercept that from an event handler... they'd still have to manually type their variable to match the chart types, and I could see that being just as hairy.

[the remark this answers is not in the document]
Comment (carloskelly13) [reply example/app/ordinal-data.tsx:30]: Yeah, the user experience is great (testing on iOS and Android) but at the cost of some developer complexities having to know about 
- SharedValues
- Using them as derived values when wanting to render
- The shape/types of their data
- Intricacies of the UI thread

**Review by carloskelly13 — APPROVED** [review]
I do like this overall, it does add complexity and another dimension, but it feels like the right approach. I'm still only slightly concerned about

`useChartPressState({ x: 0, y: { highTmp: 0 } });`

It's not clear you're passing an initial state based on the function name. Also we might want to support additional options, your call but we could do:

```tsx

useChartPressStateWithInitialValue({ x: 0, y: { highTmp: 0 } });

// or

useChartPressState({ initialValue: { x: 0, y: { highTmp: 0 } }, otherOptions: ... });

```

File: website/docs/cartesian/cartesian-chart.md [hunk 3]
```diff
@@ -54,12 +54,20 @@ An array of objects to be used as data points for the chart.
...
+:::
+
 ### `yKeys` (required)
 
 A `string[]` array of string indicating the _keys_ of each `data[number]` object to be used on the dependent (y) axis for charting. E.g. `yKeys={["lowTmp", "highTmp"]}` if you want to chart both high and low temperatures on the y-axis and those values have keys of `lowTmp` and `highTmp` respectively.
 
 This prop accepts an _array_ of strings because the `CartesianChart` supports multiple ranges (e.g. plotting both high and low temperatures), and the `CartesianChart` component needs to know about all of the slices of the dataset you plan to plot (to e.g. determine the total range of the chart).
 
+:::info
+The `yKeys` prop must point to fields that have `number` values. That is, only `number`s can be used as dependent values for charting purposes.
```
Comment (carloskelly13) [inline website/docs/cartesian/cartesian-chart.md:68]: point or map? We might want to give a quick example here on what this means code-wise.

File: website/docs/cartesian/cartesian-chart.md [hunk 4]
```diff
@@ -54,12 +54,20 @@ An array of objects to be used as data points for the chart.
 
 A `string` value indicating the _key_ of each `data[number]` object to be used on the independent (x) axis for charting. E.g. `"day"` if you want to use the `"day"` field of the data points for the x-axis.
 
+:::info
+The `xKey` prop must point to a field that has either a `number` or `string` type, as these are the values that can be reasonably serialized by Reanimated.
```
Comment (carloskelly13) [inline website/docs/cartesian/cartesian-chart.md:58]: point or map? We might want to give a quick example here on what this means code-wise.

**Pull request merged**